python/gui/auto_generated/raster/qgsrasterlayerproperties.sip.in

```diff
@@ -0,0 +1,59 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/gui/raster/qgsrasterlayerproperties.h                            *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+
+
+
+class QgsRasterLayerProperties : QgsOptionsDialogBase
+{
+%Docstring
+Property sheet for a raster map layer
+
+.. versionadded:: 3.10
+%End
+
+%TypeHeaderCode
+#include "qgsrasterlayerproperties.h"
+%End
+  public:
+
+    enum StyleType
+    {
+      QML,
+      SLD
+    };
+
+    QgsRasterLayerProperties( QgsMapLayer *lyr, QgsMapCanvas *canvas, QWidget *parent = 0, Qt::WindowFlags = QgsGuiUtils::ModalDialogFlags );
+%Docstring
+Constructor
+
+:param lyr: Map layer for which properties will be displayed
+:param canvas: the QgsMapCanvas instance
+:param parent: the parent of this widget
+:param fl: windows flag
+%End
+
+
+  public slots:
```

Comment:
This should be made private -- we don't want to expose this

Reply:
It seems SIP doesn't like when I make this slot private, I am investigating why... I've got a compilation error from a file related to SIP (sip_guipart0.cpp) that contains the method `sipQgsRasterLayerProperties::optionsStackedWidget_CurrentChanged(int a0)`.
The error message says that `QgsRasterLayerProperties::optionsStackedWidget_CurrentChanged(int)` is private. I am not sure to understand why this method is used elsewhere than in `QgsRasterLayerProperties`...

Reply:
Not expose it for python or for python and c++ API ? When I make this slot private (c++), I've got a compile error from a file related to sip even if run the sipify_all.sh script... I have to investigate why but I have difficulties to understand this sip issue...